tilavarauspalvelu/integrations/keyless_entry/typing.py

```diff
@@ -30,17 +42,97 @@ class PindoraReservationResponse(TypedDict):
     access_code_valid_minutes_before: int
     access_code_valid_minutes_after: int
     access_code_generated_at: datetime.datetime
+    access_code_is_active: bool
+    begin: datetime.datetime
+    end: datetime.datetime
+
+
+class PindoraSeasonalBookingAccessCodeValidity(TypedDict):
+    reservation_unit_id: uuid.UUID
+    access_code_valid_minutes_before: int
+    access_code_valid_minutes_after: int
+    begin: datetime.datetime
+    end: datetime.datetime
+
+
+class PindoraReservationSeriesAccessCodeValidity(TypedDict):
+    access_code_valid_minutes_before: int
+    access_code_valid_minutes_after: int
     begin: datetime.datetime
     end: datetime.datetime
 
 
+class PindoraSeasonalBookingResponse(TypedDict):
+    access_code: str
+    access_code_keypad_url: str  # url
+    access_code_phone_number: str
+    access_code_sms_number: str
+    access_code_sms_message: str
+    access_code_generated_at: datetime.datetime
+    access_code_is_active: bool
+    reservation_unit_code_validity: list[PindoraSeasonalBookingAccessCodeValidity]
+
+
+class PindoraReservationSeriesResponse(TypedDict):
+    reservation_unit_id: uuid.UUID
+    access_code: str
+    access_code_keypad_url: str  # url
+    access_code_phone_number: str
+    access_code_sms_number: str
+    access_code_sms_message: str
+    access_code_generated_at: datetime.datetime
+    access_code_is_active: bool
+    reservation_unit_code_validity: list[PindoraReservationSeriesAccessCodeValidity]
+
+
+################
+# Request data #
+################
+
+
 class PindoraReservationCreateData(TypedDict):
     reservation_id: str  # uuid
     reservation_unit_id: str  # uuid
     begin: str  # datetime
     end: str  # datetime
+    is_active: NotRequired[bool]  # default: false
```

Comment:
Does `is_active` mean that the code is immediately active, or that if it's set to False, it will never be active?

Reply:
The latter, sort of corresponds to Varaamo's tentative reservation or one requiring staff approval.

Reply:
Added note about function to comment.